packages/react-native/core/EventEmitter/NativeEventEmitter.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+'use strict';
+
+import { type EventSubscription, type IEventEmitter } from '../vendor/emitter/EventEmitter';
+import Platform from '../Utilities/Platform';
+import RCTDeviceEventEmitter from './RCTDeviceEventEmitter';
+
+export interface NativeModule {
+  addListener(eventType: string): void;
+  removeListeners(count: number): void;
+}
+
+export type { EventSubscription };
+
+/**
+ * `NativeEventEmitter` is intended for use by Native Modules to emit events to
+ * JavaScript listeners. If a `NativeModule` is supplied to the constructor, it
+ * will be notified (via `addListener` and `removeListeners`) when the listener
+ * count changes to manage "native memory".
+ *
+ * Currently, all native events are fired via a global `RCTDeviceEventEmitter`.
+ * This means event names must be globally unique, and it means that call sites
+ * can theoretically listen to `RCTDeviceEventEmitter` (although discouraged).
+ */
+export class NativeEventEmitter implements IEventEmitter {
+  _nativeModule: NativeModule;
+
+  constructor(nativeModule: NativeModule) {
+    if (Platform.OS === 'ios') {
+      if (nativeModule === null) {
+        throw new Error('`new NativeEventEmitter()` requires a non-null argument.');
+      }
+    }
```

Comment:
Worth at least collapsing the if conditions, but perhaps also doing a falsy check instead of a strict null check.
```suggestion
    if (Platform.OS === 'ios' && !nativeModule) {
      throw new Error('`new NativeEventEmitter()` requires a non-null argument.');
    }
```
But do we even need the falsy/null check? Can we rely on TypeScript, or is this being called automatically under-the-hood?

Reply:
I am not 100% sure if it's okay to remove this. As this is taken from React Native's repo, I will keep the check here for now at least.